### PR TITLE
UHF-29 Component library

### PR DESCRIPTION
**Pull request by khalima**
How to test: 
- Clone hel.fi environment (https://github.com/City-of-Helsinki/drupal-helfi) and run `make new`
- Go to theme directory `cd public/themes/custom/hdbt`
   - and switch to this branch `git fetch && git checkout --track origin/UHF-29_component_library`.
   - Compile the theme by running `nvm use && npm i && npm run build`
- Enable HDBT Component library module `drush en -y hdbt_component_library && drush cr`
- Log in to https://helfi.docker.sh and go to https://helfi.docker.sh/admin/appearance/hdbt/component-library
- Check that the component library looks some-what like the image below
image
- Check that when editing `./templates/component_library/component-library--hdbt.html.twig` the changes appear to the component library

### CODE REVIEW

File: src/scss/components/misc/_system-messages.scss [hunk 1]
```diff
@@ -4,6 +4,7 @@
 //
 .hds-notification {
   margin-bottom: var(--spacing-4-xl);
+
   &__label {
```
Comment (teroelonen) [inline src/scss/components/misc/_system-messages.scss:8]: No &___ in theme.

File: src/scss/component-library.scss [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+  transition: max-height 0.2s ease-out;
+}
+
+.component-library__code,
+.component-library__example {
+  padding: 15px;
+}
+
+.component-library__code {
+  background-color: #f3f3f3;
```
Comment (teroelonen) [inline src/scss/component-library.scss:60]: No raw values, use variables.

File: src/scss/component-library.scss [hunk 3]
```diff
@@ -0,0 +1,62 @@
+.component-library-wrapper {
+  background-color: white;
+  font-family: Arial, serif;
+  padding: 2rem;
+}
+
+.component-library__nav {
+  margin: 0 -5px;
```
Comment (teroelonen) [inline src/scss/component-library.scss:8]: Always comment absolute pixels :) This file has plenty 💋

File: src/scss/component-library.scss [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+
+  a {
+    border-right: 1px solid gray;
+    color: black;
+    font-size: 12px;
```
Comment (teroelonen) [inline src/scss/component-library.scss:13]: Use font-size mixin ;)

File: src/scss/component-library.scss [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+    padding: 3px 5px;
+    text-decoration: none;
+    text-transform: uppercase;
+
+    &:hover {
+      color: var(--color-bus-dark-50);
```
Comment (teroelonen) [inline src/scss/component-library.scss:19]: Did we agree to use this kind of variables or the regular ones?

**Review by teroelonen — CHANGES_REQUESTED** [review]
Small fixes for styles :)

File: src/scss/component-library.scss (hunk 5 above)
Comment (khalima) [reply src/scss/component-library.scss]: We did agree on SASS variables.

**Review by teroelonen — APPROVED** [review]
Good to go :)

**Pull request merged**